src/components/CommonComponents/Pagination/util/buildPaginationModel.ts

```diff
@@ -0,0 +1,148 @@
+// eslint-disable-next-line no-shadow
+export enum PageType {
+  BREAK = 'BREAK',
+  NEXT = 'NEXT',
+  NUM = 'NUM',
+  PREV = 'PREV',
+}
+
+export interface PageModel {
+  className?: string;
+  type: PageType;
+  num: number;
+  disabled?: boolean;
+  selected?: boolean;
+}
+
+export function buildPaginationModel(
```

Comment:
I still wonder if we cannot use another external NPM package for this. It is probably the most complex logic we have within the whole website. And that's not a good sign. I bet there are dozens of NPM packages that can do this part here for us 🙏

Reply:
fixed